### PR TITLE
Fix stripe after rejected payment

### PR DESCRIPTION
**Pull request by mattkrick**
fixes root cause of a null stripeSubscriptionId

TEST
- [ ] create a new org, upgrade to pro, it works
- [ ] change billing info (test cards here: https://stripe.com/docs/testing). it works

This is tedious, sorry!

TEST - FAILED PAYMENT
- run `yarn start:tunnel` requires ultrahook, check your .env & install via http://www.ultrahook.com/
- in stripeFailPayment, comment out L25 (throw Error...)
- in StripeManager, uncomment L24 (trial_end...)
- Create a new org, upgrade them with CC `4000 0000 0000 0341` with any valid expiry & cvv
- after 15 seconds, visit http://localhost:3000/me/organizations/:orgId & verify that the invoice is pending. if not pending, keep refreshing
- click "See Details" for the pending invoice, copy the invoiceId at the end of the url (looks like `in_1EgKSjFLaFINmHnBFAYC1kK4`)

- go to /admin/graphql, run the following mutation (change the invoiceId to yours):
```
mutation {
  stripeFailPayment(invoiceId: "in_1EgKSjFLaFINmHnBFAYC1kK4") {
    error {
      message
    }
  }
}
```
- [ ] refresh the org page, see that the invoice is marked as failed
- [ ] goto the team dash, see that there is modal preventing usage  
- [ ] update the CC number (`4242424242424242`)
- [ ] refresh the org billing page after 15 seconds, see 3 invoices: a failed one, a pending one, and an upcoming one.

### CODE REVIEW

**Comment by mattkrick** [comment]
@jordanh ready for review!

**Comment by mattkrick** [comment]
@jordanh friendly ping for review

**Comment by jordanh** [comment]
Looking at this one now...

**Comment by jordanh** [comment]
# Test Cycle 1

TEST
- [ ] create a new org, upgrade to pro, it works
- [ ] change billing info (test cards here: https://stripe.com/docs/testing). it works

TEST - FAILED PAYMENT
- run `yarn start:tunnel` requires ultrahook, check your .env & install via http://www.ultrahook.com/
- in stripeFailPayment, comment out L25 (throw Error...)
- in StripeManager, uncomment L24 (trial_end...)
- Create a new org, upgrade them with CC `4000 0000 0000 0341` with any valid expiry & cvv
- after 15 seconds, visit http://localhost:3000/me/organizations/:orgId & verify that the invoice is pending. if not pending, keep refreshing
- click "See Details" for the pending invoice, copy the invoiceId at the end of the url (looks like `in_1EgKSjFLaFINmHnBFAYC1kK4`)

- go to /admin/graphql, run the following mutation (change the invoiceId to yours):
```
mutation {
  stripeFailPayment(invoiceId: "in_1EgKSjFLaFINmHnBFAYC1kK4") {
    error {
      message
    }
  }
}
```
- [ ] refresh the org page, see that the invoice is marked as failed
- [ ] goto the team dash, see that there is modal preventing usage  
- [ ] update the CC number (`4242424242424242`)
- [ ] refresh the org billing page after 15 seconds, see 3 invoices: a failed one, a pending one, and an upcoming one.

**Comment by jordanh** [comment]
Notes:

### TS Warnings

```
WARNING in [at-loader] ./src/server/billing/helpers/generateInvoice.ts:126:25
    TS2352: Conversion of type '{ [x: string]: string | number | Date; id: string; amount: number; email: string; }' to type 'ReducedUnpausePartial | ReducedStandardPartial' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Property 'startAt' is missing in type '{ [x: string]: string | number | Date; id: string; amount: number; email: string; }' but required in type 'ReducedStandardPartial'.

WARNING in [at-loader] ./src/server/billing/helpers/generateInvoice.ts:255:9
    TS2322: Type 'Date' is not assignable to type 'number'.
```

**Comment by jordanh** [comment]
After adding the `STRIPE_WEBHOOK_SECRET` environment variable this worked great! Just need to clean up those TS warnings and then this is safe to merge...

File: src/server/graphql/mutations/stripeFailPayment.ts [hunk 1]
```diff
@@ -18,37 +19,34 @@ export default {
...
+    const invoice = await manager.retrieveInvoice(invoiceId)
+    const {amount_due: amountDue, customer, metadata, subscription, paid} = invoice
+    const customerId = customer as string
     let orgId = metadata.orgId
     if (!orgId) {
-      ;({
-        metadata: {orgId}
-      } = await stripe.customers.retrieve(customerId))
+      const customer = await manager.retrieveCustomer(customerId)
+      orgId = customer.metadata.orgid
```
Comment (jordanh) [inline src/server/graphql/mutations/stripeFailPayment.ts]: Paranoia check: is this the correct casing for this variable coming back from stripe?

**Pull request merged**